### PR TITLE
FEAT: Update AudioTrack functionality on mobile/tablet

### PR DESCRIPTION
**Pull request by Yura33-dev**
## Description

Button in `AudioTrack` component turn ON/OFF music when you press entire area of component at mobile and tablets.
Functionality of component at desktop has not been changed.

## Changes Made

- Added additional  styles for `playToggle` element in `AudioTrack.module.css` to make entire area clickable for tablets and mobile devices
- Provided `playToggle` element with `aria-label` to improve accessibility
- Ensured `playToggle` element is accessible with a keyboard

### CODE REVIEW

File: src/components/ui/AudioTrack/AudioTrack.module.css [hunk 1]
```diff
@@ -55,18 +55,26 @@
...
   color: inherit;
   cursor: pointer;
+
+  position: absolute;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  display: flex;
+  justify-content: flex-start;
```
Comment (naumch1k) [inline src/components/ui/AudioTrack/AudioTrack.module.css]: @Yura33-dev Per [MDN Web Docs](https://www.w3.org/TR/css-flexbox-1/#justify-content-property), `flex-start` is the initial value of the CSS Flexbox `justify-content` property. We can safely remove this line as it won't affect our element. Unless I'm missing something 🤓 ?

**Review by naumch1k — CHANGES_REQUESTED** [review]
Hey @Yura33-dev! Great job making sure the entire audio track element is clickable on mobile devices, everything works as intended and it really improves UX 🔥

I left one tiny comment below if you want to take a look.

Additionally, while testing your changes, I realized that I forgot to ensure that this button is accessible 🤦‍♀️🤫 Would you mind taking care of this while you're on this task, please?

I suggest using an `aria-label` attribute that is conditionally assigned with `Play ${name} preview` or `Pause ${name} preview`, depending on whether the track is currently playing or not and where `${name}` is a track name prop of this component.

Please re-request the review once you are done with these changes.

File: src/components/ui/AudioTrack/AudioTrack.module.css (hunk 1 above)
Comment (Yura33-dev) [reply src/components/ui/AudioTrack/AudioTrack.module.css]: @naumch1k You are absolutely right! I will remove this property

Comment (Yura33-dev) [reply src/components/ui/AudioTrack/AudioTrack.module.css]: Also I have provided `aria-label` attribute for button.

**Comment by naumch1k** [comment]
@Yura33-dev 👋 Thank you for improving keyboard accessibility by removing `display: none;` from the play toggle, which now allows our users to start and pause audio playback with the keyboard.

While testing of your changes, I found an issue where the `:focus-within` behavior was conflicting with hover effects. Specifically, after clicking to play an audio track and moving the cursor away, the AudioTrack element remained in focus, preventing the equalizer from appearing until clicking elsewhere to remove focus.

https://github.com/naumch1k/one-fall/assets/76534205/8a4cfd94-eb56-4d3d-b49d-1a81424476f5

To resolve this, I adjusted how we target the AudioTrack element when dealing with focus. Rather than targeting the root element itself, I selected the root element that has a play toggle with focus as a child. This allowed me to replace `:focus-within` instead of `:focus-visible`, ensuring our elements behave as intended.

```CSS
.root:has(.playToggle:focus-visible) .number {
  ...
}
```

Would you please test my changes to see if I didn't miss anything? If all is well and you are OK with my changes, I believe we are ready to merge 🚀

**Comment by Yura33-dev** [comment]
@naumch1k Yes, totally agree with yours changes. Now it looks pretty good. Let`s merge it!)

**Review by naumch1k — APPROVED** [review]

**Comment by naumch1k** [comment]
@Yura33-dev Done!

**Pull request merged**